src/main/java/com/mate/dao/jdbcimpl/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,257 @@
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String selectQuery = "SELECT c.id as id, "
+                + "model, "
+                + "manufacturer_id, "
+                + "m.name as manufacturer_name, "
+                + "m.country as manufacturer_country "
+                + "FROM cars c"
+                + " LEFT JOIN manufacturers m on c.manufacturer_id = m.id"
+                + " join cars_drivers cd on c.id = cd.car_id"
+                + " join drivers d on cd.driver_id = d.id"
+                + " where c.deleted = false and driver_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(selectQuery)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(parseCarFromResultSet(resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars", e);
+        }
+    }
+
+    private void updateDriversByCar(Car car) {
+        insertAllDrivers(car.getId(), car.getDrivers());
+        deleteAllDriversExceptList(car.getId(), car.getDrivers());
+    }
+
+    private void insertAllDrivers(Long carId, List<Driver> drivers) {
+        if (drivers == null || drivers.size() == 0) {
+            return;
+        }
+        StringBuilder transaction = new StringBuilder(
+                "INSERT INTO cars_drivers (car_id, driver_id) VALUES ");
+        for (int i = 0; i < drivers.size(); i++) {
+            transaction.append("(?, ?)");
+            if (i != drivers.size() - 1) {
+                transaction.append(", ");
+            }
+        }
+        transaction.append("ON DUPLICATE KEY UPDATE car_id = car_id");
+        String insertQuery = transaction.toString();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(
+                             insertQuery)) {
+            for (int i = 0; i < drivers.size(); i++) {
+                Driver driver = drivers.get(i);
+                preparedStatement.setLong((i * 2) + 1, carId);
+                preparedStatement.setLong((i * 2) + 2, driver.getId());
+            }
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert drivers " + drivers, e);
+        }
+    }
+
+    private void deleteAllDriversExceptList(Long carId, List<Driver> exceptions) {
```

Comment:
```suggestion
    private void deleteAllDrivers(Car car) {
```
let's first remove all data that relates to this car, from `cars_drivers` table
and then insert the necessary data to this table - in a different method